qa/tasks/cephadm.py

```diff
@@ -451,7 +451,11 @@ def ceph_bootstrap(ctx, config):
         # so, stop them individually
         for role in ctx.daemons.resolve_role_list(None, CEPH_ROLE_TYPES):
             cluster, type_, id_ = teuthology.split_role(role)
-            ctx.daemons.get_daemon(type_, id_, cluster).stop()
+            try:
+                ctx.daemons.get_daemon(type_, id_, cluster).stop()
+            except AttributeError:
+                log.error(f'Failed to stop {(cluster, type_, id_)} "{role}" {ctx.daemons.daemons.keys()}')
```

Comment:
This prints:

```
Failed to stop ('rgw', 'realm', 'zone.a') "rgw.realm.zone.a" dict_keys(['ceph.mon', 'ceph.mgr', 'ceph.osd', 'ceph.rgw', 'ceph.prometheus', 'ceph.node-exporter', 'ceph.alertmanager', 'ceph.grafana'])
```

which means we have `cluster == 'rgw'` which breaks this routine. 

@jdurgin or  @liewegas  any ideas? Is this a bug in `split_role`?

Reply:
I think the simplest way to make this work is to change the resolve_role_list() call to use cluster_aware=true, like the rest of the callers in cephadm.py and ceph.py.

Then you'll get 'client.rgw.realm.zone.a', which split_role() will handle correctly.